src/team_lib/mhba/xccl_mhba_collective.c

```diff
@@ -48,10 +47,16 @@ static xccl_status_t xccl_mhba_node_fanin(xccl_mhba_team_t     *team,
             ctrl_v = (int *)((ptrdiff_t)team->node.ops[index].ctrl +
                              MHBA_CTRL_SIZE * i);
             if (*ctrl_v != request->seq_num) {
-                return XCCL_INPROGRESS;
+                if (*ctrl_v == -request->seq_num){
+                    request->buff_change_flag = 1;
+                }
+                else {
+                    return XCCL_INPROGRESS;
+                }
             }
         }
     }
+    request->buff_change_flag = request->self_change_flag ? 1 : request->buff_change_flag;
```

Comment:
minor comment: i would rename "buff_change_flag" to smth that reflects more the meaning: like "need_mkey_update". It is not immediately clear what is "buffer change": it may be the same buffer pointer but different msg size and then we need to rebuild mkeys.